jose/generate.go

```diff
@@ -1,17 +1,30 @@
 package jose
 
 import (
+	"crypto"
 	"crypto/ecdsa"
 	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/rsa"
+	realx509 "crypto/x509"
 
 	"github.com/pkg/errors"
 	"github.com/smallstep/cli/crypto/pemutil"
 	"github.com/smallstep/cli/crypto/randutil"
+	"github.com/smallstep/cli/pkg/x509"
 	"golang.org/x/crypto/ed25519"
 )
 
+const (
+	jwksUsageSig = "sig"
+	jwksUsageEnc = "enc"
+)
+
+var (
+	ErrAmbiguousCertKeyUsage = errors.New("jose/generate: certificate's key usage is ambiguous, it should be for signature or encipherment, but not both (use --subtle to ignore usage field)")
+	ErrNoCertKeyUsage        = errors.New("jose/generate: certificate doesn't contain any key usage (use --subtle to ignore usage field)")
```

Comment:
Can you add comments to this variables, the linter is failing because of this.

Reply:
actually turns out we're not using them external to this package yet, so making them private for now

Reply:
Sounds good